### DISCUSSION TITLE
Writing a Haskell package and issues replicating `--standalone`

### DISCUSSION THREAD

**Discussion by patrickaldis** [jgm/pandoc#9891] [General]
I'm writing a Haskell package and currently trying to produce the same results in Haskell as running with `--standalone` in the command line.

I'm trying to produce mathjax rendered markdown. My current Haskell code is:
```Haskell
testCase = do
  let mdsrc = "$\\forall X . \\forall Y . \\left( X.XCoord =Y.XCoord \\right)\\$"
  htmlOut <- runIO $ readMarkdown rOpts (pack mdsrc) >>= writeHtml5String wOpts
  case htmlOut of
    Right f -> writeFile "out.html" $ unpack f
    Left e -> print e

wOpts :: WriterOptions
wOpts = def {writerHTMLMathMethod = MathJax defaultMathJaxURL}

rOpts :: ReaderOptions
rOpts = def {readerExtensions = pandocExtensions, readerStandalone = True}
```

Which doesn't seem to produce the header. Is the idea that `readerStandalone` replicates the functionality of `--standalone`? It's value doesn't seem to have an effect on the html output

running `pandoc -i <file> -o <output>.html --standalone --mathjax` produces the desired result

**Comment by jgm** [comment]
You need to set `writerTemplate` to a `Just` value to get standalone output.

**Reply by patrickaldis** [reply]
Oh right thanks so much. That works great now.

**Reply by jgm** [reply]
`readerStandalone` only has an effect in one or two readers, e.g. RST where in a standalone document the initial heading becomes the title metadata.  Mostly what `--standalone` does is to set `writerTemplate` to the default template for the format used.
